Sources/ClientRuntime/Networking/Streaming/StreamReader.swift

```diff
@@ -7,14 +7,31 @@ import AwsCommonRuntimeKit
 import class Foundation.FileHandle
 import class Foundation.FileManager
 
+/// Protocol for reading and writing to a stream.
 public protocol StreamReader: AnyObject {
+    /// The number of bytes that are available for reading.
     var availableForRead: UInt { get }
+
+    /// Whether or not the stream has finished writing.
+    /// This is used to determine when to stop reading from the stream.
```

Comment:
```suggestion
    /// Returns true if the stream has finished writing, otherwise returns false
```

The line `/// This is used to determine when to stop reading from the stream.` is a bit misleading here since it's an implementation detail and this is the protocol. IMO this comment should only be added to the implementation.